content/docs/hoon/reference/stdlib/5d.md

```diff
@@ -4,4 +4,318 @@ weight = 46
 template = "doc.html"
 +++
 
-(To be documented)
\ No newline at end of file
+## `++vang`
+
+Set `++vast` parameters
+
+#### Source
+
+```hoon
+++  vang                                                ::  set ++vast params
+  |=  [bug=? wer=path]                                  ::  bug: debug mode
+  %*(. vast bug bug, wer wer)                           ::  wer: where we are
+```
+
+---
+
+## `++vast`
+
+Main parsing core
+
+### Source
+
+This core is too large to include here, please refer to `hoon.hoon` for the
+source.
+
+---
+
+### `++vest`
+
+Parse hoon
+
+Parsing `rule`. Parse hoon of any form.
+
```

Comment:
a rule is a `$-(nail edge)`. I have omitted accepts/produces details for `rule`s because otherwise you'd have whole pages of "accepts: nail, produces: edge". Previous `rule` sections are also formatted that way, eg https://urbit.org/docs/hoon/reference/stdlib/4h